lychee-lib/src/types/error.rs

```diff
@@ -0,0 +1,173 @@
+use std::{any::Any, convert::Infallible, fmt::Display, hash::Hash, path::PathBuf};
```

Comment:
So we decided against using this-error? Shouldn't it avoid a lot of that boilerplate? You probably had a good reason for not using it.

Reply:
No. It's just easier to write by hand in my current workflow so I started without this-error.

It's way easy to implement `this-error` if you want to.
There's one tiny gain for writing them by hand, which is the faster compilation. It might only be faster for a few seconds at most. If that's true, consider that lychee is used in CI, it could translate to a considerable improvement overall.

Reply:
Sure, then let's keep your version and if we see that we want to move to `this-error` in the future, we can always do so. 👍